File: microservers/billetterieservice/src/main/java/io/multi/billetterieservice/domain/Commune.java
package io.multi.billetterieservice.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Commune {

    private Long communeId;
    private String communeUuid;
    private Long villeId;
    private String libelle;
    private Boolean actif;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;

    // Champs additionnels pour les jointures
    private String villeUuid;
    private String villeLibelle;
    private String regionUuid;
    private String regionLibelle;
}

File: frontend/ultima-ng-19.0.0/src/app/interface/commune.ts
export interface ICommune {
    communeId: number;
    communeUuid: string;
    libelle: string;
    actif: boolean;
    villeId: number;
    villeUuid: string;
    villeLibelle: string;
    regionUuid: string;
    regionLibelle: string;
    createdAt: Date | string;
    updatedAt: Date | string;
}

export interface ICommuneCreateRequest {
    villeUuid: string;
    libelle: string;
}

export interface ICommuneUpdateRequest {
    libelle: string;
    villeUuid?: string;
}
